### PR TITLE
[fix]: fixed deletion of records when they are of the same record type.

### PR DESCRIPTION
**Pull request by andela-oakinniranye**

### CODE REVIEW

File: pkg/recordUpdate.go [hunk 1]
```diff
@@ -4,8 +4,8 @@ package pkg
 func UpdateRecord(domain string, accessToken string, record DNSRecord, records []DNSRecord) error {
 	correctRecordExists := false
 	for _, r := range records {
-		if r.Type == record.Type {
-			if r.Name == record.Name && r.Value == record.Value+"."+domain && !correctRecordExists {
+		if r.Type == record.Type && r.Name == record.Name {
```
Comment (andela-oakinniranye) [inline pkg/recordUpdate.go]: we should only care about records matching the name and the type and deletion should only happen iff the record.value isn't the same as the existing value.
Previous implementation deleted all records matching the type, only exempting the record which has the existing name and value.
```go
r.Type == record.Type // e.g. filter all `A` records

r.Name == record.Name && r.Value == record.Value+"."+domain && !correctRecordExists // if it's the existing record, set correctRecordExists to true and don't update <- all other A records would be deleted here without nothing replacing them
```

File: pkg/recordUpdate.go [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
+)
+
 // UpdateRecord will find all current records and removes them if they are not valid. If no valid records are found it will create a new one.
 func UpdateRecord(domain string, accessToken string, record DNSRecord, records []DNSRecord) error {
 	correctRecordExists := false
 	for _, r := range records {
-		if r.Type == record.Type {
-			if r.Name == record.Name && r.Value == record.Value+"."+domain && !correctRecordExists {
+		if r.Type == record.Type && strings.HasPrefix(r.Name, record.Name) {
+			if r.Value == record.Value {
```
Comment (oreoluwa) [inline pkg/recordUpdate.go]: `r.Value` and  `record.Value` are the ip addresses. so `record.Value+"."+domain` yields `ip + example.com`, hence records would always be replaced.

**Comment by andela-oakinniranye** [comment]
cc. @oscartbeaumont please let me know your thoughts

**Comment by oscartbeaumont** [comment]
I am happy with the change. I just updated the checking logic for a more exact match (just to reduce possible edge cases). If you are happy with my change then I am happy to merge the PR.

**Pull request merged**